express-notes-api/server.ts

```diff
@@ -0,0 +1,73 @@
+import express, { Request, Response } from 'express';
+import fs from 'fs/promises';
+
+const app = express();
+
+type Note = {
+  nextId: number;
+  notes: Record<string, number>;
+};
+
+let notes: Note;
+
+async function updateNotes(): Promise<void> {
+  const d = await fs.readFile('data.json', 'utf-8');
+  notes = JSON.parse(d);
+}
+
+async function read(req: Request, res: Response): Promise<void> {
+  try {
```

Comment:
In a server, you need to re-read the data on _every_ request. This is because the data is changing in multiple places so you are never certain that the global `notes` is the most up-to-date version of the data.